DependencyInjectionContainer/DIContainer.cs

```diff
@@ -38,9 +38,27 @@ public void Register(object implementation, ServiceLifetime lifetime)
             services.Add(new Service(implementation, lifetime));
```

Comment:
Do you think it's reasonable to add constraint for allowing to register classes only?

Reply:
How could I define what class implements this interface when user register interface?